core/lib/generators/solidus/install/install_generator.rb

```diff
@@ -323,29 +344,26 @@ def detect_payment_method_to_install
       return 'paypal' if Bundler.locked_gems.dependencies['solidus_paypal_commerce_platform']
       return 'bolt' if Bundler.locked_gems.dependencies['solidus_bolt']
 
-      descriptions = {
-        paypal: "- [#{set_color 'paypal', :bold}] Install `solidus_paypal_commerce_platform` (#{set_color :default, :bold}).",
-        bolt: "- [#{set_color 'bolt', :bold}] Install `solidus_bolt`.",
-        none: "- [#{set_color 'none', :bold}] Skip installing a payment method.",
-      }
-
-      payment_methods = PAYMENT_METHODS
-
-      if @selected_frontend != 'classic'
-        payment_methods -= ['bolt']
-        descriptions.delete(:bolt)
+      selected_frontend_payment_methods = PAYMENT_METHODS.select do |payment_method|
+        payment_method[:frontends].include?(@selected_frontend)
       end
 
       selected = options[:payment_method] || (options[:auto_accept] && 'paypal') ||
         ask_with_description(
           default: 'paypal',
-          limited_to: payment_methods,
+          limited_to: selected_frontend_payment_methods.map { |payment_method| payment_method[:name] },
           desc: <<~TEXT
             Which payment method would you like to use?
 
-            #{descriptions.values.join("\n")}
+            #{selected_frontend_payment_methods.map { |payment_method| formatted_payment_method_description(payment_method) }.join("\n")}
           TEXT
         )
     end
+
+    def formatted_payment_method_description(payment_method)
```

Comment:
@waiting-for-dev I extracted `formatted_payment_method_description`. I think we can go one step further and convert the payment method into a value object class with `formatted_payment_method_description` as one of its methods, but I also think this present implementation is good enough. Your thoughts?

Reply:
Yeah, good that you mentioned that the legacy frontend will be removed soon, as it would likely simplify this class.